### ISSUE TITLE
What does "Save State" do?

### ISSUE THREAD

**Issue by SugarRayLua** [copy/v86#700]
Hello. I'm a novice Linux user and very interested in the Virtual x86 project. However, I'm having trouble figuring out what the "Save State" button on the browser does. I figured it might save the state of my browser running the Virtual x86 Linux distribution that I was running so then when I close the browser and came back to it, I could restart where I left off; however, it does not seem to do that. The "Load State" button allows me to choose a file from my own file system so I presumed that "Save State" might then also provide an option to save the current state of Linux distribution as a file but the "Save State" button doesn't seem to offer a directory choice of a place to save such a file. 

Would appreciate clarification of "Save State"'s function. 

Thank you.

**Comment by copy** [comment]
It downloads the state, so it depends on your browser where it's saved to.

**Comment by SugarRayLua** [comment]
Thanks, I see that now. :- )

**Comment by SugarRayLua** [comment]
Interestingly, on my system "Save State" only seems to work when trying to save while running one of the Demos (e.g. DSL demo); however, when I load up and run on my own .iso image that I downloaded (e.g. DSL or DSL-n .iso downloaded from internet), nothing happens when I try and "Save State".

**Comment by copy** [comment]
> however, when I load up and run on my own .iso image that I downloaded (e.g. DSL or DSL-n .iso downloaded from internet), nothing happens when I try and "Save State".

Again, I cannot reproduce this issue. When I boot up the DSL image manually, saving state works fine.

**Comment by SugarRayLua** [comment]
Thanks. I should have mentioned tested with Mac (latest OSX) desktop using Safari, Chrome and Firefox. I haven't noted any other problems with my system or browsers. I'm manually attaching DSL and DSL-N .iso to CD drive option on startup screen.

**Comment by SugarRayLua** [comment]
PS,

I think I might have figured out the problem and make the issue more reproducible to others. I forgot to mention the other variable that was different between the demo distribution I was booting with and trying to "save state" and the .iso I manually tried to boot from and save state of:  initial memory setup

The reason I chose to try manually booting from distribution .iso that I couldn't figure out how to increase the setup memory (e.g. 128 MB total memory, 8 MB video memory) when I clicked on the demo distributions. Thus, I downloaded my own .iso distributions so that I could increase to setup memory (e.g. I tried 2048 MB total memory and 128 MBG video memory) and boot from the .iso I attached to the boot CD drive. 

However, when I tried again manually booting from my distribution .iso (e.g. DSL or DSL-N) and configured the setup memory back to the defaults (e.g. 128 MB total memory, 8 MB video memory) that I was able to boot up fine on my iPad from my manual .iso and additionally was able to save the state just fine. I had thought that 2048 MB total memory and 128 MB video memory would be fine for a virtual x86 machine (iPad Pro has enough memory for that) but perhaps there is a memory configuration limit? Perhaps with the change in the memory variable others can reproduce this issue. If not, I'll know for my use of the Virtual x86 machine to keep my memory configuration lower. Thanks!
